autosklearn/data/validation.py

```diff
@@ -406,10 +406,19 @@ def _check_and_encode_target(
 
             # Mypy redefinition
             assert self.target_encoder is not None
-            self.target_encoder.fit(y)
+
+            # remove ravel warning from pandas Series
+            if len(y.shape) > 1 and y.shape[1] == 1 and hasattr(y, "values"):
```

Comment:
Is there a reason why this is only an issue for pandas series and not numpy arrays?

Reply:
Yeah, for numpy:
https://github.com/automl/auto-sklearn/blob/28305d414ee3e0656cdf8ded8f949865cb50699d/autosklearn/data/validation.py#L137

For pandas, if we ravel the values so early we lose the column/dtype info.

Reply:
Okay, got it. What does actually happen if the validator was trained with a numpy array and we then pass a pandas array or the other way round? Should we have unit tests for that?